pkg/providers/v2/cloud.go

```diff
@@ -39,6 +46,71 @@ var _ cloudprovider.Interface = (*cloud)(nil)
 
 // cloud is the AWS v2 implementation of the cloud provider interface
 type cloud struct {
+	creds  *credentials.Credentials
+	region string
+}
+
+// EC2Metadata is an abstraction over the AWS metadata service.
+type EC2Metadata interface {
+	// Query the EC2 metadata service (used to discover instance-id etc)
+	GetMetadata(path string) (string, error)
+}
+
+func getAvailabilityZone(metadata EC2Metadata) (string, error) {
+	return metadata.GetMetadata("placement/availability-zone")
+}
+
+// Derives the region from a valid az name.
+// Returns an error if the az is known invalid (empty)
+func azToRegion(az string) (string, error) {
+	if len(az) < 1 {
```

Comment:
nit: this probably reads slightly better as:
```
if len(az) == 0
```

Reply:
gotcha, updated!